cafebabel/static/js/script.js

```diff
@@ -77,65 +79,66 @@ if (!checkAutofill(input)) {
+  input.addEventListener('blur', () => input.value || deactivateInput(input))
 })
 
 /* animation title field create article */
 Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
   const h1 = inputh1.parentElement
-  inputh1.addEventListener('focus', () => {
-      h1.classList.add('active')
-  })
-  inputh1.addEventListener('blur', () => {
-      if (inputh1.value) return
-      h1.classList.remove('active')
-  })
+  inputh1.addEventListener('focus', () => h1.classList.add('active'))
+  inputh1.addEventListener('blur', () => inputh1.value || h1.classList.remove('active'))
 })
 
 /* highlight file upload area on hover or dragenter */
-function addListenerMulti(element, events, fn) {
+function addListenerMulti (element, events, fn) {
   events.split(' ').forEach(event => element.addEventListener(event, fn, false)) /* https://stackoverflow.com/questions/8796988/binding-multiple-events-to-a-listener-without-jquery */
 }
 const articleFileInput = document.querySelector('.create-article-page .file input')
 const dropArea = document.querySelector('canvas')
 if (articleFileInput) {
   addListenerMulti(articleFileInput, 'dragenter focus click', () => {
-      dropArea.classList.add('active')
+    dropArea.classList.add('active')
   })
   addListenerMulti(articleFileInput, 'dragleave blur drop', () => {
-      dropArea.classList.remove('active')
+    dropArea.classList.remove('active')
   })
   articleFileInput.addEventListener('change', () => {
-      document.querySelector('.file label').innerHTML(' ')
+    document.querySelector('.file label').innerHTML(' ')
   })
 }
 
 /* display picture file name after selection on profile edit https://tympanus.net/codrops/2015/09/15/styling-customizing-file-inputs-smart-way/ */
-const profileFileInput = document.querySelector('.profile-page main .file input')
-const profileLabel = document.querySelector('.profile-page main .file input + label')
-if (profileLabel) {
-  profileFileInput.addEventListener('change', (event) => {
-    const fileName = event.target.value.split( '\\' ).pop()
-    profileLabel.innerHTML = fileName
+const profileFileInput = document.querySelector('.profile-page.edit .file input')
+const profileLabel = document.querySelector('.profile-page.edit .file output')
+const sizeInfo = profileFileInput.parentElement.querySelector('small')
+profileFileInput && profileFileInput.addEventListener('change', (event) => {
+  const image = event.target.files[0]
+  const fileName = image.name
+  const reader = new FileReader()
+  reader.readAsDataURL(image)
+  reader.addEventListener('loadend', (event) => {
+    const maxSize = Math.round(reader.result.length / 1024) > 500
+    if (!maxSize) {
+      profileLabel.innerHTML = `
+      <figure>
+        <img src=${reader.result} alt="Preview of your profile picture">
+        <figcaption>${fileName}</figcaption>
+      </figure>
+    `
+    } else {
+      sizeInfo.classList.add('oversized')
+      profileFileInput.value = ''
```

Comment:
I think we should be more explicit here given that I can still submit the form without any update of the picture in case it's over 500Ko. Either change the message and/or block the form submission. What do you think @cafebabel-webmaster ?

Reply:
we already had the issues on _bio_ and _website_ fields : if you type something and then erase it, you can save the profile anyway. I agree it's not super UX-friendly. I will open an issue for that. Thanks for your remark, always good to hear

Reply:
Shall we take this issue into this sprint so it's "done"?

Reply:
I added to the to do list for the sprint#5